### DISCUSSION TITLE
Question regarding Registration ID for x509 DPS

### DISCUSSION THREAD

**Discussion by ghost** [Azure/azure-iot-sdk-c#2416] [Ideas]
Similar question was asked https://github.com/Azure/azure-iot-sdk-c/issues/2234

In that case as I understand, if registration ID is derived from x509 subject common name (CN) of the device certificate, why not just parse the registration ID information from the certificate itself? In the DPS sample programs the user needs to specify this value:    
https://github.com/Azure/azure-iot-sdk-c/blob/3fd808bee3c88f8578dcc30c0a6d1d396c172070/provisioning_client/samples/prov_dev_client_ll_x509_sample/prov_dev_client_ll_x509_sample.c#L23-L26  

But if my application has information about x509 certificate in a string, I'm thinking I could derive registration ID information from that variable, say from `x509certificate` as seen below but in some separate function, that would reduce the option list a user may need to input themselves.

https://github.com/Azure/azure-iot-sdk-c/blob/3fd808bee3c88f8578dcc30c0a6d1d396c172070/provisioning_client/samples/prov_dev_client_ll_x509_sample/prov_dev_client_ll_x509_sample.c#L43-L48  

The call to this new function would be something like this, it would parse base64 encoded x509 certificate presumably with the help of openssl library functions:  
```
static const char* registration_id = Get_Registration_ID_From_x509(x509certificate);
```

Would a solution like that make any sense or am I missing some key point here and should leave Registration ID to be configured separately as seen in the example?

**Comment by CIPop** [comment]
Thank you for the feedback! We will consider this as a feature request.

We have considered this in the past but decided against adding more dependencies such as an ASN.1 parser. While using OpenSSL might be simple, we have to consider support for multiple TLS stacks (mBedTLS, WolfSSL, SChannel, etc) making this feature a medium to large project.
